templates/ecs.yaml

```diff
@@ -0,0 +1,321 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Parameters:
+  KeyName:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: Name of an existing EC2 KeyPair to enable SSH access to the ECS instances.
+  VpcId:
+    Type: AWS::EC2::VPC::Id
+    Description: A VPC to run ECS instances in.
+  BastionHostSgId:
+    Type: String
+    Description: The ID of the security group for bastion hosts
+  PublicSubnetId:
+    Type: List<AWS::EC2::Subnet::Id>
+    Description: Public subnet IDs from the VPC for ALB placement. Minimum of two.
+  PrivateSubnetId:
+    Type: List<AWS::EC2::Subnet::Id>
+    Description: Private subnet IDs from the VPC for ECS node placement. Minimum of two.
+  DesiredCapacity:
+    Type: Number
+    Default: '1'
+    Description: Number of instances to launch in your ECS cluster.
+  MaxSize:
+    Type: Number
+    Default: '1'
+    Description: Maximum number of instances that can be launched in your ECS cluster.
+  MinSize:
+    Type: Number
+    Default: '1'
+    Description: Minimum number of instances that can be launched in your ECS cluster.
+  InstanceType:
+    Description: EC2 instance type
+    Type: String
+    Default: t3a.micro
+    AllowedValues: [t3.micro, t3.small, t3.medium, t3.large, t3a.micro, t3a.small, t3a.medium, t3a.large]
+    ConstraintDescription: Please choose a valid instance type.
+  DockerImageTag:
+    Type: String
+    Description: The Docker image to run from registry registry.gitlab.com/cri-iatlas/iatlas-api
+  # This is commented out as, though it is the approach recommended to get the latest AMI, it doesn't provide KeyNames fro the AMI's.
+  # For now, using AMI's from https://ami.sageit.org/ as we believe KeyNames have been generated for these.
+  # LatestAmiId:
```

Comment:
@zaro0508 and @ahayden  - You had recommended using this approach for acquiring the latest AMI Id dynamically. When we used this method, we didn't have a valid KeyName. What worked for us was creating an instance and getting a valid KeyName then creating an AMI from that instance and referencing that AMI Id. We see that https://ami.sageit.org/ has AMIs that look to be built the same way. Should we be using one of these so there is a KeyName?

Reply:
i think we need to create an ECS optimized AMI.  Would you be able to handle that @jonryser ?  This is our template for creating AMIs with packer. https://github.com/Sage-Bionetworks-IT/packer-ami-template

You will just need to base it off of a base ECS AMI from AWS.  The template will also setup the CI to build and deploy images to our AMI repo (ami.sageit.org) to be consumed by this project.

Reply:
@zaro0508 - Thanks! I will look into this.